k8s/k8s.go

```diff
@@ -1259,6 +1259,13 @@ func (k *k8sOps) ValidateDaemonSet(name, namespace string, timeout time.Duration
 			}
 		}
 
+		if ds.Status.ObservedGeneration == 0 {
```

Comment:
nit: this check should be before we call GetDaemonSetPods.

Reply:
done